elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchIndexManager.java

```diff
@@ -417,7 +424,14 @@ public void performOperations(List<LuceneWork> workList, IndexingMonitor monitor
 			elasticsearchWorks.add( luceneWork.acceptIndexWorkVisitor( visitor, monitor ) );
 		}
 
-		workProcessor.executeSyncSafe( elasticsearchWorks );
+		if ( sync ) {
+			workProcessor.executeSyncSafe( elasticsearchWorks );
+		}
+		else {
+			for ( ElasticsearchWork<?> work : elasticsearchWorks ) {
```

Comment:
I suspect this isn't dealing with the `ErrorHandler` in the usual way?

The `org.hibernate.search.exception.ErrorContext` is able to highlight a single failure our of a group, where the group usually was meant to be "all operations from the current transaction".

It's probably not easy to apply the `ErrorHandler` in the usual way as it normally would handle a failure as something like "out of operations [A, B, C] we failed on B " somehow suggesting that C wasn't executed.

In this case the various operations might have been applied out of order so you should collect all success/fail operations to forge a new kind of error state.